QualityControl/lib/services/Status.service.js

```diff
@@ -0,0 +1,136 @@
+
+const QC_VERSION_EXEC_COMMAND = 'yum info o2-QualityControl | awk \'/Version/ {print $3}\'';
+
+/**
+ * Service for retrieving information and status of QCG dependencies
+ */
+export class StatusService {
+  /**
+   * Setup StatusService constructor and initialize needed dependencies
+   * @param {object} packageInfo - object containing partial information from package.json file
+   * @param {object} config - partial configuration of QCG setup
+   */
+  constructor(packageInfo, config = {}) {
+    this._logger = new Log(`${process.env.npm_config_log_label ?? 'qcg'}/status-service`);
+
+    /**
+     * @type {CcdbService}
+     */
+    this._dataService = undefined;
+
+    /**
+     * @type {ConsulService}
+     */
+    this._onlineService = undefined;
+
+    this._packageInfo = packageInfo;
+    this._config = config;
+  }
+
+  /**
+   * Method that builds an object with some information about the server itself
+   * @returns {object} - info on version, host and port where application is deployed
+   */
+  retrieveOwnStatus() {
+    return {
+      status: { ok: true },
+      version: this._packageInfo?.version ?? '-',
+    };
+  }
+
+  /**
+   * Send back info about the framework
+   * @returns {object} - object containing status and framework information
+   */
+  async retrieveFrameworkInfo() {
+    return {
+      qcg: this.retrieveOwnStatus(),
+      qc: this.retrieveQcVersion(),
+      ccdb: await this.retrieveDataServiceStatus(),
+      consul: {
+        status: await this.retrieveOnlineServiceStatus(),
+      },
+    };
+  }
+
+  /**
+   * Retrieve data service (CCDB) status and issue if any
+   * @returns {Promise<{object}>} - status of the data service
+   */
+  async retrieveDataServiceStatus() {
+    try {
+      const { version } = await this._dataService.retrieveVersion();
+      return { status: { ok: true }, version };
+    } catch (err) {
+      return { status: { ok: false, message: err.message || err } };
+    }
+  }
+
+  /**
+   * Retrieve status of the online service (Consul) if it was configured and issue if any
+   * @returns {Promise<Resolve, Reject>} - status of the online service
+   */
+  async retrieveOnlineServiceStatus() {
+    if (!this._onlineService) {
+      return { ok: false, message: 'Live Mode was not configured' };
+    }
+    try {
+      await this._onlineService.getConsulLeaderStatus();
+      return { ok: true };
+    } catch (err) {
+      return { ok: false, message: err.message || err };
+    }
+  }
+
+  /**
+   * Method to execute QC version retrieval command
+   * @returns {string} - version of QC deployed on the system
+   */
+  retrieveQcVersion() {
+    let version = 'Not part of an FLP deployment';
+    if (this._config.qc?.enabled) {
+      try {
+        version = execSync(QC_VERSION_EXEC_COMMAND, { timeout: 3000 }).toString().trim();
+      } catch (error) {
+        this._logger.errorMessage(error, { level: 99, system: 'GUI', facility: 'status-service' });
```

Comment:
You still return `ok` here?

Reply:
Yes, for now it is better to return `ok` because QCG only benefits of 2 states. `green`(success) or `red` (error) and there is no `grey(invalid/not-configured)` like in AliECS GUI. This will come at a later stage.
This is to prevent oncallers being asked to fix an error that it actually does not exist